From 78d08b80c07c8a144ca1f18d0c5df9a22952926b Mon Sep 17 00:00:00 2001
From: Susant Sahani <145210+ssahani@users.noreply.github.com>
Date: Thu, 3 May 2018 20:50:56 +0530
Subject: [PATCH] ethtool: get_glinksettings Fix copy (#8889)

It should be other way around.

(cherry picked from commit 801d2c9f5de04fdc7e95a0dafc771903923f0304)
---
 src/udev/net/ethtool-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udev/net/ethtool-util.c b/src/udev/net/ethtool-util.c
index 9bdaef8d90..8179a5e2f8 100644
--- a/src/udev/net/ethtool-util.c
+++ b/src/udev/net/ethtool-util.c
@@ -418,7 +418,7 @@ static int get_glinksettings(int fd, struct ifreq *ifr, struct ethtool_link_uset
         if (!u)
                 return -ENOMEM;
 
-        ecmd.req = u->base;
+        u->base = ecmd.req;
 
         offset = 0;
         memcpy(u->link_modes.supported, &ecmd.link_mode_data[offset], 4 * ecmd.req.link_mode_masks_nwords);
-- 
2.17.0

